services/sources/whoisxml.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2017 Jeff Foley. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package sources
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/OWASP/Amass/config"
+	eb "github.com/OWASP/Amass/eventbus"
+	"github.com/OWASP/Amass/net/http"
+	"github.com/OWASP/Amass/requests"
+	"github.com/OWASP/Amass/resolvers"
+	"github.com/OWASP/Amass/services"
+)
+
+// WhoisXML is the Service that handles access to the WhoisXML data source.
+type WhoisXML struct {
+	services.BaseService
+
+	API        *config.APIKey
+	SourceType string
+	RateLimit  time.Duration
+}
+
+//Reverse handles response json
+type Reverse struct {
```

Comment:
Remember this is a global namespace with all of the other services. Might want to pick a more descriptive name like `WhoisXMLResponse`